habana/ubi8-python-3.8/Dockerfile

```diff
@@ -0,0 +1,171 @@
+    ssh-keygen -A && \
+    mkdir -p /var/run/sshd && echo "/usr/sbin/sshd -p 3022" | tee -a ~/.bashrc
+
+RUN echo "export LANG=en_US.UTF-8" >> /root/.bashrc
+RUN export LANG=en_US.UTF-8
+ENV GC_KERNEL_PATH=/usr/lib/habanalabs/libtpc_kernels.so
+ENV HABANA_LOGS=/var/log/habana_logs/
+ENV HABANA_SCAL_BIN_PATH=/opt/habanalabs/engines_fw
+ENV HABANA_PLUGINS_LIB_PATH=/opt/habanalabs/habana_plugins
+
+## Install habana pytorch
+ENV LANG=en_US.UTF-8
+ENV PYTHONPATH=/root:/usr/lib/habanalabs/
+
+RUN dnf install -y \
+    curl \
+    cairo-devel \
+    numactl-devel \
+    iproute \
+    which \
+    zlib-devel \
+    lapack-devel \
+    openblas-devel \
+    numactl \
+    gperftools-devel && \
+    dnf clean all && rm -rf /var/cache/yum
+
+RUN wget --no-verbose https://"${ARTIFACTORY_URL}"/artifactory/gaudi-pt-modules/"${VERSION}"/"${REVISION}"\
+/pytorch/rhel86/pytorch_modules-v"${PT_VERSION}"_"${VERSION}"_"${REVISION}".tgz && \
+    mkdir /root/habanalabs /root/habanalabs/pytorch_temp && \
+    tar -xf pytorch_modules-v"${PT_VERSION}"_"${VERSION}"_"${REVISION}".tgz -C /root/habanalabs/pytorch_temp/. && \
+    pip3 install mpi4py==3.1.4 && \
+    pip3 install /root/habanalabs/pytorch_temp/*.whl && \
+    pip3 install $(grep "pytorch-lightning" /root/habanalabs/pytorch_temp/requirements-pytorch.txt) && \
+    pip3 install -U habana-lightning-plugins=="${VERSION}"."${REVISION}.1" && \
+    /sbin/ldconfig && \
+    echo "source /etc/profile.d/habanalabs.sh" >> ~/.bashrc && \
+    pip3 uninstall -y pillow && \
+    pip3 uninstall -y pillow-simd && \
+    pip3 install pillow-simd==7.0.0.post3 && \
+    rm -rf /root/habanalabs/pytorch_temp/ && \
+    rm -rf pytorch_modules-v"${PT_VERSION}"_"${VERSION}"_"${REVISION}".tgz
+ENV LD_PRELOAD=/lib64/libtcmalloc.so.4
+ENV TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=7516192768
+
+RUN dnf clean all && rm -rf /var/cache/dnf && rm -rf /tmp/*
+
+## Label the image with details required by ODH
+LABEL name="odh-notebook-habana-jupyter-datascience-ubi8-python-3.8" \
+    summary="Jupyter HabanaAI datascience notebook image for ODH notebooks" \
+    description="Jupyter HabanaAI datascience notebook image with base Python 3.8 builder image based on ubi8 for ODH notebooks" \
+    io.k8s.display-name="Jupyter HabanaAI datascience notebook image for ODH notebooks" \
+    io.k8s.description="Jupyter HabanaAI datascience notebook image with base Python 3.8 builder image based on ubi8 for ODH notebooks" \
+    authoritative-source-url="https://github.com/opendatahub-io/notebooks" \
+    io.openshift.build.commit.ref="main" \
+    io.openshift.build.source-location="https://github.com/opendatahub-io/notebooks/tree/main/habana/ubi8-python-3.8" \
+    io.openshift.build.image="quay.io/opendatahub/workbench-images:habana-jupyter-datascience-ubi8-python-3.8"
+
+# Fix permissions to support pip in Openshift environments
+RUN chmod -R g+w /opt/app-root/lib/python3.8/site-packages && \
+    fix-permissions /opt/app-root -P
```

Comment:
Shall we add here instructions to replace the notebook's launcher, "(ipykernel)" with Python's version 3.x to keep UI consistency? like the following:
```
sed -i -e "s/Python.*/$(python --version | cut -d '.' -f-2)\",/" /opt/app-root/share/jupyter/kernels/python3/kernel.json && \
```